Documentation for redesigned sidechains (#1516)
XRPLF/xrpl-dev-portal: content/concepts/interoperability/cross-chain-bridges.md

---
html: cross-chain-bridges.html
parent: xrpl-interoperability.html
blurb: Cross-chain bridges for the XRP Ledger enable value in the form of XRP and other tokens (IOUs) to move efficiently between blockchains.
labels:
  - Blockchain
---
# Cross-Chain Bridges

Cross-chain bridges for the XRP Ledger enable value in the form of XRP and other tokens (IOUs) to move efficiently between blockchains such as the XRP Ledger and its sidechains.

A locking chain is a blockchain that holds assets that are then put into trust when a bridge to an issuing chain is created.

An issuing chain is an independent ledger with its own consensus algorithm and transaction types and rules. It acts as its own blockchain.

> Both the locking and issuing chains operate as parllel networks with independent nodes and validators. They rely on independent [witness servers](witness-server.html) to watch transactions between the two chains and attest that assets have moved into specifically designated accounts.

Reviewer: ```suggestion
Both the locking and issuing chains operate as parallel networks with independent nodes and validators. They rely on independent [witness servers](witness-server.html) to watch transactions between the two chains and attest that assets have moved into specifically designated accounts.
```

## Terminology

> * Bridge: A method of moving assets/value from one blockchain to another.

Reviewer: Should the term be bolded/somehow emphasized in comparison to the description?

* Witness: Independent servers that are aware of the locking and issuing chains. See [witness servers](witness-server.md) for more information.

* Cross-chain transfer: A transfer of assets from one chain to another.

* Source chain: The chain that a cross-chain transfer begins from. The transfer is from the source chain and to the destination chain.

* Destination chain: The chain that a cross-chain transfer ends at. The transfer is from the source chain and to the destination chain.

* Locking chain: The chain on which the bridge locks and unlocks assets.

* Issuing chain: The chain on which the bridge mints and burns assets.

* Door account: A special type of account that is used to move assets from one chain to another. The door account on a locking chain is used to put assets into trust, and the door account on an issuing chain used to issue wrapped assets. 

* Cross-chain claim ID: A special identifier used for cross-chain transfers. A cross-chain claim ID represents *one* cross-chain transfer of value.

## How Do Cross-Chain Transactions Work?

> ### Pre-requisites

Reviewer: ```suggestion
### Prerequisites
```

You must ensure that the following are set up and running before initiating cross-chain transactions. 

> * Ensure that the witness server(s) are up and running.

Reviewer: You might want to add that the sidechain needs to be set up with validators and running

* Set up a bridge between the two chains, including a _door account_ on each chain. On one chain, the asset is locked and unlocked, hence the name "locking chain", and on the other chain, assets are minted and burned, or issued and reclaimed, hence the name "issuing chain".
* When using the XRP Ledger Mainnet as one of the chains, ensure that  the `Sidechains` amendment is enabled in the `rippled.cfg` configuration file. 

> ### Working of Cross-Chain Transactions 

Reviewer: The grammar here feels weird - maybe something like this:
```suggestion
### How A Cross-Chain Transfer Works
```

> At a high-level, cross-chain transactions involve the following steps: 

Reviewer: This flow is good, but perhaps there should also be links to the actual transactions submitted.

1. Claim a cross-chain claim ID on the issuing chain.

> 2. Submit a cross-chain transfer transaction on the locking chain, attaching the claimed cross-chain claim ID and include a reward for the witness servers. This locks the asset on the locking chain.

Reviewer: ```suggestion
2. Submit a commit transaction on the locking chain, attaching the claimed cross-chain claim ID and include a reward for the witness servers. This locks the asset on the locking chain.
```

3. Obtain the attestations from the witness servers that the transaction occurred on the issuing chain.

> 4. Submit a transaction claim for the transferred value on the issuing chain, attaching the attestation as proof that the value was indeed transferred.

Reviewer: The claim only needs to be submitted if something goes wrong (a trustline isn't set up properly, deposit auth is set up, etc.)

The [`xrpl-sidechain-cli`](https://github.com/XRPLF/sidechain-cli) is a commandline tool that simplifies setting up a cross-chain bridge and issuing chains on your local machine. 

Follow the [tutorial](https://github.com/XRPLF/sidechain-cli/blob/main/scripts/tutorial.sh) to walk through the steps of creating a bridge and completing your first cross-chain transaction. 

## XRPL Custom Network Explorer 

The XRP Ledger Explorer provides a way to look up historical transactions, accounts, ledgers, fees, exchange rates, timestamps, sequence numbers, node uptime, IP addresses, topology, versions and peers for the XRP Ledger mainchain. 

Similarly, you can use the [XRP Ledger Custom Network Explorer](https://custom.xrpl.org/) to connect an XRP Ledger Sidechain (_issuing chain_) and look up network information. 

## See Also

- [Witness Server](witness-server.html)
- [Tutorial](https://github.com/XRPLF/sidechain-cli/blob/main/scripts/tutorial.sh)
- [Transaction Reference](transaction-types.html)
    - [xchaincreateaccountcommit][]
    - [xchaincreatebridge][]
    - [xchaincreateclaimid][]
    - [xchaincommit][]
    - [xchainaddattestation][]
    - [xchainclaim][]

>     - [xchainmodifybridge][]

Reviewer 2: Transaction names are case-sensitive.

```suggestion
    - [XChainCreateAccountCommit][]
    - [XChainCreateBridge][]
    - [XChainCreateClaimID][]
    - [XChainCommit][]
    - [XChainAddAttestation][]
    - [XChainClaim][]
    - [XChainModifyBridge][]
```

Furthermore, these reference links won't work unless you add these names to `content/_snippets/tx-type-links.md`.

4. When there are enough signatures to reach quorum, the XRP is automatically released on the issuing chain to the destination account. In some cases, for example a trustline is not set up properly, or a deposit auth is set up, you need to submit a transaction claim for the transferred value on the issuing chain, attaching the attestation as proof that the value was indeed transferred.
6. The rewards are then distributed to the witness servers' accounts on the issuing chain.

Consider an example where Alice wants to send XRP from her account on the XRP Ledger Mainnet to her account sAlice on a sidechain. In this example. the XRP Ledger Mainnet is the locking chain and the sidechain is the issuing chain. 

For the sake of this example, let's assume that the following prerequisites have been met.

* The sidechain has been configured with validators and is up and running.
* A bridge that transfers XRP between the locking chain and the issuing chain has been set up with `XChainCreateBridge`.
* Witness servers are up and running.
* Alice has already run a `XChainCreateAccountCommit` transaction and has an account on the sidechain.

> <!-- Add image of just the bridge created-->

Reviewer 2: I think this comment can be removed since there's an image right after this?

![Cross-chain Transactions](img/xrpl-bridging-solution.png "Cross-chain transactions")

> 1. sAlice first checks out a claim ID with `XChainCreateClaimID` on the sidechain, specifying the above bridge. She retrieves the claim ID from the transaction metadata or the `xchaincreateclaimid` RPC call.

Reviewer 2: [Avoid the term "RPC call"](https://github.com/XRPLF/xrpl-dev-portal/blob/master/CONTRIBUTING.md#terminology). Use "API method" instead. Also, looks like there's a typo ("sAlice").

```suggestion
1. Alice first checks out a claim ID with `XChainCreateClaimID` on the sidechain, specifying the above bridge. She retrieves the claim ID from the transaction metadata or the `xchaincreateclaimid` API method.
```

When the `xchaincreateclaimid` API method has docs, this would also be a good place to link them.

---
html: cross-chain-bridges.html
parent: interoperability.html
blurb: Cross-chain bridges for the XRP Ledger enable value in the form of XRP and other tokens (IOUs) to move efficiently between blockchains.
labels:
  - Blockchain
---
# Cross-Chain Bridges

> Cross-chain bridges for the XRP Ledger enable value in the form of XRP and other tokens (IOUs) to move efficiently between blockchains such as the XRP Ledger and its sidechain.

Reviewer 2: - sidechains plural, probably
- seems like a good place to link "tokens" for those who need the background info
- Maybe another good piece of background reading to link is the parallel networks page

```suggestion
Cross-chain bridges for the XRP Ledger enable value in the form of XRP and other [tokens](tokens.html) to move efficiently between blockchains such as the XRP Ledger and its sidechains.
```

A locking chain is a blockchain that holds assets that are then put into trust when a bridge to an issuing chain is created.

> An issuing chain is an independent ledger with its own consensus algorithm and transaction types and rules. It acts as its own blockchain.

Reviewer 2: These definitions are a bit hard to conceptualize with only this much context. Would the Mainnet qualify as one, the other, or both? Some brief examples might help.

Reviewer: It's explained a little more succinctly/clearly in the `Terminology` section, but it's a little confusing that it's in both places.

Both the locking and issuing chains operate as parllel networks with independent nodes and validators. They rely on independent [witness servers](witness-servers.md) to watch transactions between the two chains and attest that assets have moved into specifically designated accounts.

## How Do Cross-Chain Transactions Work?

> *Pre-requisites*

Reviewer 2: Rather than italics, should this be a sub-heading?

You must ensure that the following are set up and running before initiating cross-chain transactions. 

> * The witness server(s) are up and running.

Reviewer 2: The other bullet points here are in imperative voice; this one isn't. Is that intentional?

> * Set up a bridge between the two chains. This inovles initializing the bridge on both chains. Each chain has a _door account_ that controls that end of the bridge on-chain. On one chain, the asset is locked and unlocked (hence the name "locking chain"), and on the other chain, assets are minted and burned, or issues and reclaimed (hence the name "issuing chain").

Reviewer 2: - typo ("inovles")
- Do bridges to multiple other chains share a single door (on each chain)? or does each separate bridge have its own door account? I'm thinking it's an "each bridge" thing.
- Feels redundant to say setting up a bridge involves initializing it.
- I think splitting the naming explanations into two sentences, and removing parentheses, makes it read just a little better.
- typo ("are issues")

```suggestion
* Set up a bridge between the two chains, including a _door account_ on each chain. On one chain, the asset is locked and unlocked, hence the name "locking chain". On the other chain, assets are minted and burned, or issued and reclaimed, hence the name "issuing chain".
```

> * When using the XRP Ledger mainchain as one of the chains, ensure that  the `Sidechains` amendment is enabled in the rippled.cfg configuration file. 

Reviewer 2: - I think this is referring to the [testing amendments](https://xrpl.org/amendments.html#testing-amendments) thing, but there's some context missing to make this an actionable step. Maybe the intent here is that all servers in one chain (or both?) need to have the amendment force-enabled this way? Or maybe the intended message is that you can't bridge to the Mainnet until the Sidechains amendment goes live there?
    - As an aside, the section on "Testing Amendments" could use some updates: it should remind you to restart; also, maybe it should draw a distinction between "amendments" that are open for voting and "features" that you can force-enable but are _not_ open for voting. (NegativeUNL was a "feature" in several releases before it got an amendment.)
- Use the proper noun "Mainnet" rather than "mainchain" which is not defined in the Parallel Networks doc.

> *  

Reviewer 2: Stray bullet point?

At a high-level, cross-chain transactions involve the following steps: 

> 1. Claim a cross-chain claim ID on the issuing chain.

Reviewer 2: The concept of a "cross-chain ID" hasn't been introduced yet, let alone how you claim one, so this is confusing. Is there a limited supply of IDs? Is this a per-transaction thing? Is this related to a transaction ID (hash) or a network ID?

> 2. Submit a cross-chain transfer transaction on the locking chain, attaching the claimed cross-chain claim ID and include a reward for the witness servers. This locks or burns the asset on the issuing chain.

Reviewer 2: The phrasing "locks or burns on the issuing chain" is confusing here since earlier you make a distinction wherein the _locking chain_ locks/unlocks assets but the _issuing chain_ issues/reclaims or mints/burns the asset.

> 3. Obtain the attestations from the witness servers that the transaction occurred on the issuing chain.

Reviewer 2: Should this say "locking chain"? In general, the flow of which transfers happen in which order on which chains is very unclear in these steps, which makes me think there's at least one typo using the wrong name for a chain.

Reviewer: I think it's a weird grammatical thing. The attestations are on the issuing chain, but the event they're attesting happened on the locking chain.
The witness servers also auto-send the attestations now, so you don't need to retrieve them from the witness servers anymore.

4. Submit a transaction claim for the transferred value on the issuing chain, attaching the attestation as proof that the value was indeed transferred.

> 5. The rewards are then distributed to the witness servers' accounts on the issuing chain.

Reviewer 2: passive voice: distributed _by whom_? Presumably the door accounts do the distributing?

Reviewer: It happens automatically when the transfer occurs

Consider an example where Alice wants to send XRP from her account on the XRP Ledger mainchain (locking chain) to her account sAlice on a sidechain (issuing chain).

* A bridge that transfers XRP between the XRP Ledger mainchain and the sidechain has been set up with XChainCreateBridge.
* Witness servers are up and running.

> * Alice has also already run a XChainCreateAccountCommit transaction at some point, so she has an account on the sidechain.

Reviewer 2: Whoa, what is this XChainCreateAccountCommit transaction? That wasn't mentioned in the prerequisites or the high-level overview, was it?

Reviewer: Perhaps another flow should be account creation.

<!-- Add image of just the bridge created-->

![Cross-chain Transactions](img/xrpl-bridging-solution.png "Cross-chain transactions")

> 1. sAlice first checks out a claim ID with XChainCreateClaimID on the sidechain, specifying the above bridge. She retrieves the claim ID from the transaction metadata or the TBD RPC call.

Reviewer 2: I'm guessing that "TBD RPC" is something that needs to get updated before publishing? Maybe take that out for the initial draft?

2. Alice then takes the cross-chain claim ID from sAlice’s XChainCreateClaimID transaction and submits a XChainCommit transaction on the mainchain with that claim ID, locking up a specified amount of XRP. She specifies sAlice’s account in the OtherChainDestination field.

> 3. The witnesses then take note of the XChainCommit transaction and submit XChainAddAttestation transactions on the sidechain, attesting to the fact that the XChainCommit transaction did in fact occur on the mainchain.

Reviewer 2: Should this be using "locking chain" and "issuing chain" instead of "mainchain" and "sidechain" respectively?

4. When there are enough XChainAddAttestation signatures to reach quorum, the XRP is automatically released on the sidechain to sAlice’s account.

5. If the XRP is not automatically released, for whatever reason (such as Alice forgetting to specify sAlice’s account in the OtherChainDestination field), then sAlice submits a XChainClaim transaction on the sidechain, specifying her account as the destination. This then releases the XRP on the sidechain to sAlice’s account.

## Terminology

* Bridge: A method of moving assets/value from one blockchain to another.

* Witness: Independent servers that are aware of the locking and issuing chains. See [witness servers](witness-server.md) for more information.

* Cross-chain transfer: A transfer of assets from one chain to another.

* Source chain: The chain that a cross-chain transfer begins from. The transfer is from the source chain and to the destination chain.

> * Destination chain: The chain that a cross-chain transfer ends at. The transfer is from the source chain and to the destination chain.

Reviewer 2: The phrase "ends at" makes me wonder if you can have cross-chain transfers that involve more than two chains. Can you do sidechain → mainchain → other sidechain for example? (I mean all in one and not just with multiple manual steps between)

* Locking chain: The chain on which the bridge locks and unlocks assets.

* Issuing chain: The chain on which the bridge mints and burns assets.

> * Door account: A special type of account that is used to move assets from one chain to another. The door account on a locking chain is used to put assets into trust, and the door account on an issuing chain used to issue wrapped assets. 

Reviewer 2: I believe "into trust" and "locked" are supposed to mean more or less the same thing, so it's probably best not to mix and match them here.

## Transactions

### Bridge Control: XChainModifyBridge

> This transaction modifies a Bridge ledger object on one of the chains that the bridge connects. You can only change the SignaturesReward or MinAccountCreateAmount values, because changing the bridge itself (either door account or either currency) would essentially render all cross-chain funds useless - you’d be better off creating another bridge instead. This transaction must be sent by the door account, and correctly signed using whatever signer list set it has.

Reviewer 2: You can tighten up the language and provide background reading re:multi-signing in the last sentence here:

```suggestion
This transaction modifies a Bridge ledger object on one of the chains that the bridge connects. You can only change the SignaturesReward or MinAccountCreateAmount values, because changing the bridge itself (either door account or either currency) would essentially render all cross-chain funds useless - you’d be better off creating another bridge instead. This transaction must be sent by the door account, and correctly signed using its [multi-signing](multi-signing.html) list.
```

### Cross-Chain Transfer: XChainCreateClaimID

This transaction checks out a cross-chain claim ID that is used for a cross-chain transfer. It is submitted on the destination chain, not the source chain. This is the first step of a cross-chain transfer of value. 

> A cross-chain claim ID essentially represents one cross-chain transfer of value. 

Reviewer 2: I think "represents" is not quite the word here, because it could give the impression that claiming the ID performs some of the transfer. "Identifies" is perhaps too close to defining "ID" with itself, so maybe "tracks" is better? Also, I like changing the word order so that "one transfer" is together, since that's the important part.

```suggestion
A cross-chain claim ID tracks one transfer of value across chains. 
```

### Cross-Chain Transfer: XChainCommit

> This transaction initiates a cross-chain transfer of value. This is done on the source chain, and locks/burns the value (“commits” the value), so that the equivalent amount can be minted/unlocked on the destination chain. Essentially, it tells the witness servers that the value was locked/burned. This value is tied to a specific cross-chain claim ID (which is included in the transaction).

Reviewer 2: Again, using "locks/burns" interchangeably feels weird when the definitions of the chains makes a distinction between the chains based on which happens where.

The account that owns the cross-chain claim ID on the destination chain is the account that controls the funds on the other end of the bridge. The funds go to the destination account specified in the XChainCommit transaction, if specified. If the destination account is not specified, then the claim ID owner must submit an XChainClaim transaction to determine where the funds will go on the destination chain.

### Cross-Chain Transfer: XChainAddAttestation

This transaction is submitted on the destination chain, by the witness server or anyone with access to the signatures from the witness server. It is a proof that an event (essentially just a locking/burning of funds) happened on the source chain.

> When enough witnesses have submitted their proofs on the destination chain that an event has occurred, the funds will be released to the destination account in the XChainCommit transaction, if specified. Otherwise, the claim ID owner must submit an XChainClaim transaction to determine where the funds will go on the destination chain.

Reviewer 2: Fix passive voice/future tense:

```suggestion
When enough witnesses have submitted their proofs on the destination chain that an event has occurred, the door account on the destination chain releases the funds to the destination account in the XChainCommit transaction, if specified. Otherwise, the claim ID owner must submit an XChainClaim transaction to release the funds on the destination chain.
```

### Cross-Chain Transfer: XChainClaim

> This transaction completes a cross-chain transfer of value. It allows a user to claim the value on the destination chain - the equivalent of the value locked on the source chain. A user can only claim the value if they own the cross-chain claim ID associated with the value locked on the source chain (the Account field). The user can send the funds to anyone (the Destination field). This transaction is only needed if an OtherChainDestination is not specified in the XChainCommit transaction, or if something goes wrong with the automatic transfer of funds.

Reviewer 2: Actual emdash. Also, maybe continue to use "assets" rather than changing terms to "value" at this point?

```suggestion
This transaction completes a cross-chain transfer. It allows a user to claim the assets on the destination chain—the equivalent of the assets locked on the source chain. The user must own the cross-chain claim ID associated with the assets locked on the source chain (the `Account` field). The user can send the funds to anyone (the `Destination` field). This transaction is only needed if an OtherChainDestination is not specified in the XChainCommit transaction, or if something goes wrong with the automatic transfer of funds.
```

P.S. the `Account` is always the sender of the transaction; does this mean that the `Account` and `Destination` fields referenced here are on different transactions (one on the source chain, one on the destination chain)? It feels like this sentence says that the cross-chain claim ID is in the `Account` field, but that field should contain an account address, not a unique-per-transfer ID.

### Other: XChainCreateAccountCommit

> This transaction creates an account on the sidechain. In order to start a cross-chain transfer, you need an account on the destination chain, which is a bit of a paradox situation - so to avoid that, this is a special transaction that can create an account on the destination chain for you.

Reviewer 2: - actual emdash
- "in order to"→"to"
- "a bit of a paradox situation" feels a little casual/vague. To my understanding it's not impossible to fund an account on a sidechain without this transaction, but you would need to have someone who already has funds on the sidechain to do that for you. I tried one rewording but I'm open to better ideas on this.

```suggestion
This transaction creates an account on the sidechain. To start a cross-chain transfer, you need an account on the destination chain. Since it can be hard to create an account unless you already hold assets on the destination chain, this is a special transaction that can create an account on the destination chain for you.
```

## How to Set Up a Sidechain? 

The [`sidechain_cli`](https://github.com/XRPLF/sidechain-cli) is a commandline tool that simplifies setting up bridges and issuing chains on your local machine. 

Follow the [tutorial](https://github.com/XRPLF/sidechain-cli/blob/main/scripts/tutorial.sh) to walk through the steps of creating a bridge and completing your first cross-chain transaction. 

> ## <TBD> XRPL Sidechain Explorer 

Reviewer 2: Don't forget to update this before merge.

> The XRP Ledger Explorer provides a way to look up historical transactions, accounts, ledgers, fees, exchange rates, timestamps, sequence numbers, node uptime, IP addresses, topology, versions and peers for the XRP Ledger mainchain. 

Reviewer 2: ```suggestion
The [XRP Ledger Explorer](https://livenet.xrpl.org) provides a way to look up various information on the XRP Ledger Mainnet or known testnets: you can find historical transactions, accounts, ledgers, fees, exchange rates, timestamps, sequence numbers, node uptime, IP addresses, topology, versions and peers. 
```

---
html: cross-chain-bridges.html
parent: xrpl-interoperability.html
blurb: Cross-chain bridges for the XRP Ledger enable value in the form of XRP and other tokens (IOUs) to move efficiently between blockchains.
labels:
  - Blockchain
---
# Cross-Chain Bridges

Cross-chain bridges for the XRP Ledger enable value in the form of XRP and other tokens (IOUs) to move efficiently between blockchains such as the XRP Ledger and its sidechains.

A locking chain is a blockchain that holds assets that are then put into trust when a bridge to an issuing chain is created.

An issuing chain is an independent ledger with its own consensus algorithm and transaction types and rules. It acts as its own blockchain.

Both the locking and issuing chains operate as parllel networks with independent nodes and validators. They rely on independent [witness servers](witness-servers.md) to watch transactions between the two chains and attest that assets have moved into specifically designated accounts.

## Terminology

* Bridge: A method of moving assets/value from one blockchain to another.

> * Witness: Independent servers that are aware of the locking and issuing chains. See [witness servers](witness-server.md) for more information.

Reviewer 2: Linking via `.md` like this won't work

```suggestion
* Witness: Independent servers that are aware of the locking and issuing chains. See [witness servers](witness-server.html) for more information.
```

* Destination chain: The chain that a cross-chain transfer ends at. The transfer is from the source chain and to the destination chain.

* Locking chain: The chain on which the bridge locks and unlocks assets.

* Issuing chain: The chain on which the bridge mints and burns assets.

* Door account: A special type of account that is used to move assets from one chain to another. The door account on a locking chain is used to put assets into trust, and the door account on an issuing chain used to issue wrapped assets. 

* Cross-chain claim ID: A special identifier used for cross-chain transfers. A cross-chain claim ID represents *one* cross-chain transfer of value.

## How Do Cross-Chain Transactions Work?

### Pre-requisites

You must ensure that the following are set up and running before initiating cross-chain transactions. 

* Ensure that the witness server(s) are up and running.
* Set up a bridge between the two chains, including a _door account_ on each chain. On one chain, the asset is locked and unlocked, hence the name "locking chain", and on the other chain, assets are minted and burned, or issued and reclaimed, hence the name "issuing chain".

> * When using the XRP Ledger Mainnet as one of the chains, ensure that  the `Sidechains` amendment is enabled in the `rippled.cfg` configuration file. 

Reviewer 2: It's still unclear which servers' config files should have the `Sidechains` amendment (force-?)enabled

Reviewer: The locking chain and issuing chain (the two chains the bridge is between) must both have the `XChainBridge` amendment enabled

3. Obtain the attestations from the witness servers that the transaction occurred on the issuing chain.
4. Submit a transaction claim for the transferred value on the issuing chain, attaching the attestation as proof that the value was indeed transferred.
5. The rewards are then distributed to the witness servers' accounts on the issuing chain.

Consider an example where Alice wants to send XRP from her account on the XRP Ledger Mainnet (locking chain) to her account sAlice on a sidechain (issuing chain).

* A bridge that transfers XRP between the XRP Ledger Mainnet and the sidechain has been set up with XChainCreateBridge.
* Witness servers are up and running.
* Alice has also already run a XChainCreateAccountCommit transaction at some point, so she has an account on the sidechain.

<!-- Add image of just the bridge created-->

![Cross-chain Transactions](img/xrpl-bridging-solution.png "Cross-chain transactions")

1. sAlice first checks out a claim ID with XChainCreateClaimID on the sidechain, specifying the above bridge. She retrieves the claim ID from the transaction metadata or the `xchaincreateclaimid` RPC call.

> 2. Alice then takes the cross-chain claim ID from sAlice’s XChainCreateClaimID transaction and submits a XChainCommit transaction on the Mainnet with that claim ID, locking up a specified amount of XRP. She specifies sAlice’s account in the OtherChainDestination field.

Reviewer 2: This step is the first of a few cases that refer to "Mainnet" (which traditionally means the one production XRP Ledger) whereas the previous explanations described things in terms of an "issuing chain" and a "locking chain". I believe that the thought is that Mainnet would most often be the locking chain for production sidechains, but other chains could be the locking chain in some configurations especially in test cases.

This should probably be consistent with the use of "locking chain" / "issuing chain" rather than Mainnet / sidechain, I think.

Reviewer: Agreed - I think it's fine to say "mainnet is the locking chain, sidechain is the issuing chain" at the top, and then just stick to locking chain/issuing chain terminology
